code/API_definitions/Traffic_Influence.yaml

```diff
@@ -641,28 +654,29 @@ components:
             - 'error'
             - 'deletion in progress'
             - 'deleted'
-        trafficFilters:
-          type: array
-          items:
-            type: string
-            description: Indicates the packet filters of the IP flow. The source
-             IP is not required. It is retrived by the platoform according to
-             the TI API request. If no Traffic Influece resourse is created with
-             a specific Device, any IP will be routed to the Application. If a
-             Traffic Influece resource exists with a specific Device configured,
-             only the related IPs will be routed to the local instance of the
-             Application. The destination IP is not required,it is already known
-             by the platform thanks to the appId and appInstanceId parameters
-             that are used to retive the destination IP. The protocol must be
-             specified and it is identified by a string according to the column
-             “Keyword” as defined by IANA (https://www.iana.org/assignments/\
-             protocol-numbers/protocol-numbers.xhtml), e.g. UDP or TCP.
-             The destination port must be specified.
-            example: TCP 5060
-            minItems: 1
-          minItems: 1
-          description: Identifies IP packet filters. To be used when a the
-           Application needs a traffic flow towards a specific EAS interface
+        sourceTrafficFilters:
+          description: Ports used locally by the device for flows to which
+           the requested traffic influence should apply. If omitted, then the
+           traffic influence will apply to all flows between the device and the
+           specified application server address and ports.
+          type: object
+          properties:
+            sourcePort:
+              allOf:
+                - $ref: "#/components/schemas/PortsSpec"
+        destinationTrafficFilters:
+          description: Identifies the destination IP packet filters. To be
+           used when it is needed a traffic flow towards a specific EAS
+           interface identified by a protocol and a port.  Also the Protocol
+           (e.g. TCP or UDP) can be specified.
+          type: object
+          properties:
```

Comment:
In general the port and IP address are used together for identification of specific traffic flow. Is there a need to include the IP address of device or EAS in source and destination traffic filters?

Reply:
the IP address for the source is already in Device. For the destination, we are currently just targeting application deplyed by the OP so, the destination IP is already known by the Transformation function as a consequence of AppId or AppInstanceId.

We are discussing in OPAG about this: "Support for External EAS". If we move forward with that requirement, the destination IP is required

Reply:
I think the observation is partly correct but then i see another point that if the application has multiple components then how would AppId or AppInstanceId.can be used to determine the right component if the components are having different IPs? Any suggestion or thought in such scenario?

Reply:
do you mean an Application that is exposing different IPs, Ports and maybe protocols, and the Service Provider wants to influence just some of them? Currently this is not supported. Anyway this can be a step toward the support for external EAS. even more, if we implement this, we also have the support for external EAS for free :-)

Reply:
Sure I will open a discussion in App LCM and link it here.

Reply:
Following discussion has been opened, https://github.com/camaraproject/EdgeCloud/discussions/264